### PR TITLE
Make reactive AnnotationMapper, MailboxAnnotationListener, IMAP GetMetadata & SetMetadata processor 

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
Ci error
```
java.net.ConnectException: maxPings exceeded: 2. Giving up. The clamd daemon seems not to be running
```

Maybe it was fixed in https://github.com/apache/james-project/pull/1512

I will rebase latest master

**Comment by chibenwa** [comment]
--
 

<a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/">Test Result</a> (8 failures  / ±0)<ul style="list-style-type: none; margin: 0;"><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAnnotationsByKeysShouldThrowExceptionWhenDoesNotLookupMailbox/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAnnotationsByKeysShouldThrowExceptionWhenDoesNotLookupMailbox</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAllAnnotationsShouldReturnStoredAnnotation/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAllAnnotationsShouldReturnStoredAnnotation</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAllAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAllAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAnnotationsByKeysShouldRetrieveStoreAnnotationsByKey/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAnnotationsByKeysShouldRetrieveStoreAnnotationsByKey</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAllAnnotationsShouldReturnEmptyForNonStoredAnnotation/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAllAnnotationsShouldReturnEmptyForNonStoredAnnotation</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/updateAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.updateAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/updateAnnotationsShouldCallAnnotationMapperToDeleteAnnotation/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.updateAnnotationsShouldCallAnnotationMapperToDeleteAnnotation</a></li><li style="" class="hidden"><a href="https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/updateAnnotationsShouldCallAnnotationMapperToInsertAnnotation/">org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.updateAnnotationsShouldCallAnnotationMapperToInsertAnnotation</a></li></ul>[Test Result](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/) (8 failures / ±0)

    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAnnotationsByKeysShouldThrowExceptionWhenDoesNotLookupMailbox](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAnnotationsByKeysShouldThrowExceptionWhenDoesNotLookupMailbox/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAllAnnotationsShouldReturnStoredAnnotation](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAllAnnotationsShouldReturnStoredAnnotation/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAllAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAllAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAnnotationsByKeysShouldRetrieveStoreAnnotationsByKey](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAnnotationsByKeysShouldRetrieveStoreAnnotationsByKey/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.getAllAnnotationsShouldReturnEmptyForNonStoredAnnotation](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/getAllAnnotationsShouldReturnEmptyForNonStoredAnnotation/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.updateAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/updateAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.updateAnnotationsShouldCallAnnotationMapperToDeleteAnnotation](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/updateAnnotationsShouldCallAnnotationMapperToDeleteAnnotation/)
    [org.apache.james.mailbox.store.StoreMailboxManagerAnnotationTest.updateAnnotationsShouldCallAnnotationMapperToInsertAnnotation](https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1510/4/testReport/junit/org.apache.james.mailbox.store/StoreMailboxManagerAnnotationTest/updateAnnotationsShouldCallAnnotationMapperToInsertAnnotation/)

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerAnnotationTest.java [hunk 1]
```diff
@@ -105,46 +110,49 @@ void updateAnnotationsShouldThrowExceptionWhenDoesNotLookupMailbox() {
         when(mailboxMapper.findMailboxByPath(eq(mailboxPath))).thenReturn(Mono.error(new MailboxException()));
 
         assertThatThrownBy(() -> annotationManager.updateAnnotations(mailboxPath, session, ImmutableList.of(PRIVATE_ANNOTATION)))
-            .hasCauseInstanceOf(MailboxException.class);
+            .isInstanceOf(MailboxException.class);
```
Comment (vttranlina) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreMailboxManagerAnnotationTest.java:113]: Why?
- In `MailboxManagerTest`, we assert by `isInstanceOf` API -> we should keep similar for both
- The original of this test is `isInstanceOf`, and after commit 5ccf47c38a651653a79283f9f83b491d72b852e9 (JAMES-3167 Reactify MailboxMapper), we change it to `hasCauseInstanceOf`, So IMO the revert lit is not a big deal

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Squash please

**Pull request merged**